flake.nix

```diff
@@ -48,25 +48,32 @@
           };
 
         lib = rec {
-          setActivate = base: activate: pkgs.buildEnv {
-            name = ("activatable-" + base.name);
-            paths = [
-              base
-              (pkgs.writeTextFile {
-                name = base.name + "-activate-path";
-                text = ''
-                  #!${pkgs.runtimeShell}
-                  ${activate}
-                '';
-                executable = true;
-                destination = "/deploy-rs-activate";
-              })
-            ];
-          };
 
-          nixosActivate = base: setActivate base "$PROFILE/bin/switch-to-configuration switch";
+          setActivate = builtins.trace
+            "deploy-rs#lib.setActivate is deprecated, use activate.noop, activate.nixos or activate.custom instead"
+            activate.custom;
+
+          activate = {
+            custom = base: activate: pkgs.buildEnv {
+              name = ("activatable-" + base.name);
+              paths = [
+                base
+                (pkgs.writeTextFile {
+                  name = base.name + "-activate-path";
+                  text = ''
+                    #!${pkgs.runtimeShell}
+                    ${activate}
+                  '';
+                  executable = true;
+                  destination = "/deploy-rs-activate";
+                })
+              ];
+            };
 
-          noopActivate = base: setActivate base ":";
+            nixos = base: setActivate base "$PROFILE/bin/switch-to-configuration switch";
+
+            noop = base: setActivate base ":";
```

Comment:
doesn't this mean any usage of `activate.{nixos, noop}` is going to produce a deprecation warning?